### PR TITLE
Add README.md's

### PR DESCRIPTION
**Pull request by sonjaleo**
## Description
Adding README.md files to all notebooks and data generated

## Todos
  - [x] modify README.md in /notebooks/custom_kinfraglib/
  - [x] modify REAMDE.md in  /data
  - [x] add README.md in /data/fragment_library_custom_filtered
  - [x] add README.md in data/filters
  - [x] add README.md in data/filters/Brenk
  - [x] add README.md in data/filters/retrosynthesizability
  - [x] add README.md in notebooks

## Status
- [x] Ready to go

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,8 +21,11 @@ This repository holds the following resources:
 
 1. Fragment library data and a link to the combinatorial library data.
 2. *Quick start* notebook explaining how to load and use the fragment library.  
-3. Notebooks covering the full analyses regarding the fragment and combinatorial libraries as described in 
-the corresponding paper.  
+3. Notebooks 
+
+    3.1. Notebooks covering the full analyses regarding the fragment and combinatorial libraries as described in 
+          the corresponding paper.  
+    3.2. Notebooks providing a custom filtering framework to reduce the fragment library size.
```
Comment (AndreaVolkamer) [inline README.md]:
```suggestion
    3.2. WIP: Notebooks providing a custom filtering framework to reduce the fragment library size.
```

File: README.md [hunk 2]
```diff
@@ -47,6 +50,8 @@ Following this approach, a fragment library is created with respective subpocket
 an in-depth analysis of the chemical space of known kinase inhibitors, and can be used to enumerate recombined 
 fragments in order to generate novel potential inhibitors.
 
+Custom KinFragLib provides a pipeline to custom filter KinFragLib utilizing unwanted substructures (PAINS and Brenk et al.), drug-likeness (Rule of Three and QED), synthesizability (buyable building blocks and SYBA) and pairwise retrosynthesizability. Each filter can be (de-)activated and the parameters can be modified by the user to create a custom filtered library.
```
Comment (AndreaVolkamer) [inline README.md]:
```suggestion
WIP: Custom KinFragLib provides a pipeline to filter the fragments in KinFragLib checking for unwanted substructures (PAINS and Brenk et al.), lead-/drug-likeness (Rule of Three and QED), synthesizability (similarity to buyable building blocks and SYBA) and pairwise retrosynthesizability. Each filter can be (de-)activated and the parameters can be modified by the user to create a customized filtered fragment library. 
```

File: data/README.md [hunk 3]
```diff
@@ -2,8 +2,10 @@
...
-- `fragment_library_reduced/`: Reduced fragment library: Select a diverse set of fragments (per subpocket) for recombination starting from the filtered fragment library.
+- `fragment_library/`: The full fragment library resulting from the KinFragLib fragmentation procedure comprises about 3000 fragments, which are the basis for exploring the subpocket-based chemical space of ligands co-crystallized with kinases.
+- `fragment_library_filtered/`: Filtered fragment library: Select fragments tailored for the recombination (remove pool X,  deduplicate per subpocket, remove unfragmented ligands, remove all fragments that connect only to pool X, keep only fragment-like fragments, and filter for hinge-like AP fragments).
+- `fragment_library_reduced/`: Reduced fragment library: Select a diverse set of fragments (per subpocket) for recombination, starting from the filtered fragment library.
 - `combinatorial_library/`: Combinatorial library based on the reduced fragment library.
 - `external/`: Data from external resources.
+-  `filters`: Data used for custom filters.
```
Comment (AndreaVolkamer) [inline data/README.md]:
```suggestion
- `filters`: Data used for custom filters.
```

File: data/README.md [hunk 4]
```diff
@@ -2,8 +2,10 @@
...
+- `fragment_library_custom_filtered`: Custom filtered fragment library: Pre-filtered (removed unfragmented ligands, ligands only connecting to pool X, fragments in pool X, duplicates), filtered for unwanted substructures (PAINS and Brenk), filtered for drug-likeness (Ro3 and QED), filtered for synthesizability (buyable building blocks and SYBA) and filtered for pairwise retrosynthesizability (using ASKCOS).
```
Comment (AndreaVolkamer) [inline data/README.md]: above pre-filtered contained the step 'deduplicate per subpocket', here 'duplicates', if it refers to the same step, name them consistently

Comment (AndreaVolkamer) [inline data/README.md]:
```suggestion
- `fragment_library_custom_filtered`: Custom filtered fragment library: Pre-filtered (removed unfragmented ligands, ligands only connecting to pool X, fragments in pool X, duplicates), and filtered for unwanted substructures (PAINS and Brenk), drug-likeness (Ro3 and QED), synthesizability (buyable building blocks and SYBA) and pairwise retrosynthesizability (using ASKCOS).
```

File: data/filters/Brenk/README.md [hunk 5]
```diff
@@ -0,0 +1,3 @@
+# Brenk et al.
+
+- `unwanted_substructures.csv`: file with unwanted substructures provided by Brenk et al. [(Chem. Med. Chem. (2008), 3, 535-44)](https://chemistry-europe.onlinelibrary.wiley.com/doi/full/10.1002/cmdc.200700139) containing the name and the SMARTS string of the unwanted substructure.
```
Comment (AndreaVolkamer) [inline data/filters/Brenk/README.md]:
```suggestion
- `unwanted_substructures.csv`: File with unwanted substructures provided by Brenk et al. [(Chem. Med. Chem. (2008), 3, 535-44)](https://chemistry-europe.onlinelibrary.wiley.com/doi/full/10.1002/cmdc.200700139) containing the name and the SMARTS string of the unwanted substructure.
```

File: data/filters/README.md [hunk 6]
```diff
@@ -0,0 +1,5 @@
+# Filters Data
+
+- `Brenk`: folder containing the 'unwanted_substructures.csv' provided by Brenk et al.
```
Comment (AndreaVolkamer) [inline data/filters/README.md]:
```suggestion
- `Brenk`: Folder containing the 'unwanted_substructures.csv' provided by Brenk et al.
```

File: data/filters/README.md [hunk 7]
```diff
@@ -0,0 +1,5 @@
...
+- `DataWarrior`: folder containing the 'Enamine_Building_Blocks.sdf' file created with DataWarrior.
```
Comment (AndreaVolkamer) [inline data/filters/README.md]:
```suggestion
- `DataWarrior`: Folder containing the 'Enamine_Building_Blocks.sdf' file created with DataWarrior.
```

File: data/filters/README.md [hunk 8]
```diff
@@ -0,0 +1,5 @@
...
+- `retrosynthesizability`: folder containing 'retro.txt' where all retrosynthetic pathways that were and will be requested are stored
```
Comment (AndreaVolkamer) [inline data/filters/README.md]:
```suggestion
- `retrosynthesizability`: Folder containing 'retro.txt' in which all retrosynthetic pathways that were and will be requested are stored.
```

Comment (AndreaVolkamer) [reply data/filters/README.md]: what means 'will be requested' here? Is this file not static?

File: data/filters/retrosynthesizability/README.md [hunk 9]
```diff
@@ -0,0 +1,3 @@
+# retrosynthesizability data
+
+- `retro.txt`: file containing the results that are already requested for fragment pairs and will be used to store further requested results. It contains \[pair SMILES\]; \[child(ren) 1 SMILES\]; \[child(ren) 2 SMILES\]; \[plausibility/ies\] for every requested fragment pair.
```
Comment (AndreaVolkamer) [inline data/filters/retrosynthesizability/README.md]:
```suggestion
- `retro.txt`: File containing the results that are already requested for fragment pairs and will be used to store further requested results. It contains \[pair SMILES\]; \[child(ren) 1 SMILES\]; \[child(ren) 2 SMILES\]; \[plausibility/ies\] for every requested fragment pair.
```

Comment (AndreaVolkamer) [reply data/filters/retrosynthesizability/README.md]: Not entirely sure what you mean by '... results that are already requested... ' here?
And the word 'requested' appears 4 times in the sentence. Consider rephrasing.

File: notebooks/README.md [hunk 10]
```diff
@@ -0,0 +1,3 @@
+# notebooks
+- `custom_kinfraglib`: Customizable filtering functionality to reduce fragment library size
```
Comment (AndreaVolkamer) [inline notebooks/README.md]:
```suggestion
- `custom_kinfraglib`: WIP: Customizable filtering functionality to reduce the size of the fragment library.
```

File: notebooks/custom_kinfraglib/README.md [hunk 11]
```diff
@@ -1,13 +1,22 @@
...
 ### `1_2_custom_filters_drug_likeness.ipynb`
 This notebook filters out fragments not fulfilling the Rule of Three and the Quantitative Estimate
 of Druglikeness (QED), which both reflect the molecular properties of the fragments.
 ### `1_3_custom_filters_synthesizability.ipynb`
-This notebook is filtering the fragments for syntheszibalility using a buyable building block
+This notebook is filtering the fragments for synthesizability using a buyable building block
 filter and the SYnthetic Bayesian Accessibility (SYBA).
+### `1_4_custom_filters_pairwise_retrosynthesizability.ipynb`
+This notebook is building fragment pairs using only those fragments that passed all custom filtering step.
+Afterwards, it will start a request using the ASKCOS API to check if a one step retrosynthetic route for this pair can be found.
```
Comment (AndreaVolkamer) [inline notebooks/custom_kinfraglib/README.md]:
```suggestion
Next, it requests retrosynthesis information using the ASKCOS API to check if a one-step retrosynthetic route for this pair can be found.
```

File: notebooks/custom_kinfraglib/README.md [hunk 12]
```diff
@@ -1,13 +1,22 @@
...
+Finally, it will compare the children retrieved from ASKCOS with the fragments building the pair. If the fragments are
```
Comment (AndreaVolkamer) [inline notebooks/custom_kinfraglib/README.md:15]: maybe explain what children refers to in this context

File: notebooks/custom_kinfraglib/README.md [hunk 13]
```diff
@@ -1,13 +1,22 @@
...
+substructures of the children, then they pass this filter.
+## 2. Custom filtering pipeline
+### `2_1_custom_filters_pipeline.ipynb`
+This notebook compares all the filtering steps explained before and gives the possibility to easily (de-)activate
```
Comment (AndreaVolkamer) [inline notebooks/custom_kinfraglib/README.md]: Does it 'compare' all filtering steps? Or apply all filtering steps?

File: notebooks/custom_kinfraglib/README.md [hunk 14]
```diff
@@ -1,13 +1,22 @@
...
+single filtering steps and modify the parameters.
+### `2_2_custom_filters_analysis.ipynb`
+This notebook is analyzing the filtered fragment library and comparing it with the pre-filtered and the reduced set.
```
Comment (AndreaVolkamer) [inline notebooks/custom_kinfraglib/README.md]:
```suggestion
This notebook is analyzing the custom filtered fragment library and comparing it with the pre-filtered and the reduced fragment set from the previous study.
```

File: data/fragment_library_custom_filtered/README.md [hunk 15]
```diff
@@ -0,0 +1,15 @@
+# Custom filtered fragment library
+
+The (full) fragment library resulting from the KinFragLib fragmentation procedure comprises of 7486 fragments, which are the basis for exploring the subpocket-based chemical space of ligands co-crystallized with kinases (see `data/fragment_library/`).
```
Comment (AndreaVolkamer) [inline data/fragment_library_custom_filtered/README.md]:
```suggestion
The (full) fragment library resulting from the KinFragLib fragmentation procedure comprises 7486 fragments, which are the basis for exploring the subpocket-based chemical space of ligands co-crystallized with kinases (see `data/fragment_library/`).
```

File: data/fragment_library_custom_filtered/README.md [hunk 16]
```diff
@@ -0,0 +1,15 @@
...
+
+To reduce the fragment library size and enable the recombination avoiding the recombinatorial explosion we offer the possibility to filter the fragment library by customizable filtering steps, namely:
```
Comment (AndreaVolkamer) [inline data/fragment_library_custom_filtered/README.md]:
```suggestion
To reduce the fragment library size and enable the recombination avoiding combinatorial explosion and to increase the chance of synthesizability of the newly created molecules, the fragment library can now be filtered by customizable filtering steps, namely:
```

File: data/fragment_library_custom_filtered/README.md [hunk 17]
```diff
@@ -0,0 +1,15 @@
+# Custom filtered fragment library
```
Comment (AndreaVolkamer) [inline data/fragment_library_custom_filtered/README.md]:
```suggestion
# WIP: Custom filtered fragment library
```

File: data/fragment_library_custom_filtered/README.md [hunk 18]
```diff
@@ -0,0 +1,15 @@
...
+
+The (full) fragment library resulting from the KinFragLib fragmentation procedure comprises of 7486 fragments, which are the basis for exploring the subpocket-based chemical space of ligands co-crystallized with kinases (see `data/fragment_library/`).
+
+To reduce the fragment library size and enable the recombination avoiding the recombinatorial explosion we offer the possibility to filter the fragment library by customizable filtering steps, namely:
+
+1. Pre-filtering (Remove pool X, deduplicate, remove unfragmented fragments, remove fragments only connecting to pool X and fragments in Pool X) \[not optional\]
```
Comment (AndreaVolkamer) [inline data/fragment_library_custom_filtered/README.md]:
```suggestion
1. Pre-filtering (Remove pool X, deduplicate, remove unfragmented fragments, remove fragments only connecting to pool X and fragments in pool X) \[not optional\]
```

Comment (AndreaVolkamer) [reply data/fragment_library_custom_filtered/README.md]: What means '[not optional\]' here?

File: data/fragment_library_custom_filtered/README.md [hunk 19]
```diff
@@ -0,0 +1,15 @@
...
+2. Filter for unwanted substructures (PAINS and Brenk et al.) \[optional\]
+3. Filter for drug likeness (Ro3 and QED) \[optional\]
+4. Filter for synthesizability (Buyable building blocks and SYBA) \[optional\]
+5. filter for pairwise retrosynthesizability (using ASKCOS) \[optional\]
```
Comment (AndreaVolkamer) [inline data/fragment_library_custom_filtered/README.md]:
```suggestion
5. Filter for pairwise retrosynthesizability (using ASKCOS) \[optional\]
```

File: data/fragment_library_custom_filtered/README.md [hunk 20]
```diff
@@ -0,0 +1,15 @@
...
+
+- `custom_filter_results.csv`: file containing the filtering results from the analysis applied, including the fragments SMILES and the fragments subpocket as indices and the values and boolean columns for each fragment (from the pre-filtered library) retrieved by the filtering steps.
```
Comment (AndreaVolkamer) [inline data/fragment_library_custom_filtered/README.md]:
```suggestion
- `custom_filter_results.csv`: File containing the filtering results, including per fragment, from the pre-filtered library, the SMILES and subpocket as indices, the calculated scores and boolean columns, if a fragment passes a specific filter, generated by the filtering steps.
```

File: data/fragment_library_custom_filtered/README.md [hunk 21]
```diff
@@ -0,0 +1,15 @@
...
+
+Please refer to the notebook `notebooks/custom_kinfraglib/2_1_custom_filters_pipeline.ipynb` to check how the data was generated and/or to generate your own custom fragment library (de-)activating filters and modifying the filters parameters.
```
Comment (AndreaVolkamer) [inline data/fragment_library_custom_filtered/README.md]:
```suggestion
Please refer to the notebook `notebooks/custom_kinfraglib/2_1_custom_filters_pipeline.ipynb` to check how the data was generated and/or to generate your own custom fragment library (de-)activating filters and modifying the filtering parameters.
```

**Review by AndreaVolkamer — APPROVED** [review]
@sonjaleo well done. Added only a few suggestions.

File: data/filters/README.md (hunk 8 above)
Comment (sonjaleo) [reply data/filters/README.md]: No, it's not static.
If one keeps more fragments in the previous filtering steps and new ASKCOS queries are started, the results will also be stored there.

File: data/fragment_library_custom_filtered/README.md (hunk 18 above)
Comment (sonjaleo) [reply data/fragment_library_custom_filtered/README.md]: The pre-filtering cannot be deactivated. It is always applied.

File: data/README.md (hunk 4 above)
Comment (sonjaleo) [reply data/README.md]: done

File: data/filters/retrosynthesizability/README.md (hunk 9 above)
Comment (sonjaleo) [reply data/filters/retrosynthesizability/README.md]: It contains all the results from ASKCOS, that I have already requested.

File: data/fragment_library_custom_filtered/README.md (hunk 18 above)
Comment (AndreaVolkamer) [reply data/fragment_library_custom_filtered/README.md]: ok then let's write [mandatory] instead

File: data/filters/README.md (hunk 8 above)
Comment (AndreaVolkamer) [reply data/filters/README.md]: ok, then change to: 
- `retrosynthesizability`: Folder containing 'retro.txt' in which all retrosynthetic pathways that were previously requested (or will potentially be assembled in future queries) are stored.

File: data/filters/retrosynthesizability/README.md (hunk 9 above)
Comment (AndreaVolkamer) [reply data/filters/retrosynthesizability/README.md]: ok maybe like this:
- `retro.txt`: File containing the results that were already assembled in previous queries for fragment pairs or those that will be queried in future searches. It contains \[pair SMILES\]; \[child(ren) 1 SMILES\]; \[child(ren) 2 SMILES\]; \[plausibility/ies\] for every requested fragment pair.

**Pull request merged**